### PR TITLE
Fix bucket creation for remote_state for gcs

### PR DESCRIPTION
**Pull request by brimstone**
I wasn't able to convince terragrunt v0.19.25 to create a bucket in GCS for storing remote state. It kept saying:
```
Missing required GCS remote state configuration project
```
I tracked this down to the bit where the remote_state config {} section is cleaned up BEFORE the bucket is created.

This PR fixes it in my limited testing. I'm not sure how to run only the GCS tests so pointers there would be appreciated.

### CODE REVIEW

File: remote/remote_state_gcs.go [hunk 1]
```diff
@@ -82,6 +82,11 @@ func (gcsInitializer GCSInitializer) NeedsInitialization(remoteState *RemoteStat
 		return true, nil
 	}
 
+	// Delete custom AFTER the bucket is created GCS labels that are only used in Terragrunt config and not in Terraform's backend
```
Comment (yorinasub17) [inline remote/remote_state_gcs.go]:
```suggestion
	// Delete custom GCS labels that are only used in Terragrunt config and not in Terraform's backend AFTER the bucket is created 
```

**Comment by yorinasub17** [comment]
This test `TestGCSConfigValuesEqual/equal-ignore-gcs-labels` failed:
```
    --- FAIL: TestGCSConfigValuesEqual/equal-ignore-gcs-labels (0.00s)
        remote_state_gcs_test.go:110: 
            	Error Trace:	remote_state_gcs_test.go:110
            	Error:      	Not equal: 
            	            	expected: true
            	            	actual  : false
            	Test:       	TestGCSConfigValuesEqual/equal-ignore-gcs-labels
```

**Comment by brimstone** [comment]
I think the test is fixed. I don't see a machine confirming.

**Comment by yorinasub17** [comment]
Just kicked off a build!

**Comment by yorinasub17** [comment]
@robmorgan can you also take a look to double check the functionality? The change makes sense to me, but since you wrote the original code, definitely could use another set of eyes here.

**Comment by brimstone** [comment]
Anything I can do to help merge this?

**Comment by brikis98** [comment]
@robmorgan ^

**Comment by robmorgan** [comment]
Hi all, sorry about the wait. I'll take a look tomorrow.

**Comment by brimstone** [comment]
@robmorgan Thanks! Not tomorrow is ok too, if it doesn't fall off your radar.

**Comment by robmorgan** [comment]
Okay, all the tests seem to pass locally for me. Let me check once more on CircleCI.

**Comment by robmorgan** [comment]
Released in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.20.3

**Pull request merged**